Add type hints

kasium opened the pull request:
Please note that a upgrade of black was needed due https://github.com/psf/black/issues/2964

Closes #103

kasium commented:
@lafrech after fighting with the tests for some time I think it's the best if load_operations_from_docstring would accept a None for a docstring. It is capable to handle it and an assert would break existing applications

kasium commented:
@lafrech I just picking up on long running PRs. Are you still willing to consider this feature? I would then update the code and try to make the tests green

lafrech commented:
Sorry about the lack of feedback. Thanks for the ping.

Yes, this contribution is welcome.

Thanks.

-- 
Jérôme

kasium commented:
@lafrech there is an error in Azure
```
##[warning]An image label with the label ubuntu-18.04 does not exist.
,##[error]The remote provider was unable to process the request.
```

lafrech commented:
Yep. Let's go on with #108 first.

lafrech commented:
You may rebase this now that CI is fixed.

Thanks !

kasium commented:
@lafrech sorry for the radio silence from my side. CI is now green 😄

lafrech commented on src/apispec_webframeworks/bottle.py:
```diff
@@ -37,11 +38,11 @@ class BottlePlugin(BasePlugin):
 
     @staticmethod
-    def _route_for_view(app, view):
+    def _route_for_view(app: Bottle, view):
```
Missing return type

lafrech reviewed (approved):
Thanks. LGTM.

I spotted a missing return value and you could comment on why the "ignore type" lines, but it's already much better that nothing so I'm willing to merge this.

kasium commented:
I removed all type ignores beside one where I added a comment explaining it.
To simplify the code, I removed support for the 2017 released tornado 4.4

lafrech commented:
Thanks!
